src/backlight/backlight.c

```diff
@@ -389,11 +389,13 @@ int main(int argc, char *argv[]) {
                 r = read_one_line_file(saved, &value);
                 if (r < 0) {
 
-                        if (r == -ENOENT)
-                                return EXIT_SUCCESS;
+                        if (r != -ENOENT) {
+                                log_error_errno(r, "Failed to read %s: %m", saved);
+                                return EXIT_FAILURE;
+                        }
 
-                        log_error_errno(r, "Failed to read %s: %m", saved);
-                        return EXIT_FAILURE;
+                        /* not saved: fake "0" to trigger brightness clamping if necessary */
+                        value = strdup("0");
```

Comment:
needs OOM check. also, I think it would be nicer to rearrange this block to remove one level of indentation. Specifically:

```c
r = read_one_line_file(saved, &value);
if (r == -ENOENT) {
        /* … */
        value = strdup("0");
        if (!value) {
                log_oom();
                return EXIT_FAILURE;
        }
 } else if (r < 0)  {
        …
}

Reply:
hmm, that all said, this will break keyboard backlights, i.e. devices of the "leds" subsystem (in contrast to "backlight"). This tool covers both. For kbd backlights we don't apply clamping currently which means your code would turn them off automatically, and I don't think that's intended...